engine/remote-components/src/main/java/org/datacleaner/components/remote/RemoteTransformer.java

```diff
@@ -61,25 +67,33 @@
     private String componentDisplayName;
     private String username;
     private String password;
-    private String tenant;
-    private OutputColumns cachedOutputColumns;
+
+    private OutputColumns lastOutputColumns;
+    private CreateInput lastCreateInput;
+
     private ComponentRESTClient client;
     private Map<String, Object> configuredProperties = new TreeMap<>();
 
-    public RemoteTransformer(String baseUrl, String componentDisplayName, String tenant, String username, String password) {
+    public RemoteTransformer(String baseUrl, String componentDisplayName, String username, String password) {
         this.baseUrl = baseUrl;
         this.username = username;
         this.password = password;
-        this.tenant = tenant;
         this.componentDisplayName = componentDisplayName;
     }
 
-    public void init() {
-        logger.debug("Initializing '{}' @{}", componentDisplayName, this.hashCode());
-        client = new ComponentRESTClient(baseUrl, username, password);
+    @Initialize
+    public void initClient() throws RestrictedFunctionalityException {
+        try {
+            logger.debug("Initializing '{}' @{}", componentDisplayName, this.hashCode());
+            client = new ComponentRESTClient(baseUrl, username, password);
+        } catch (Exception e) {
+            throw new RestrictedFunctionalityException(
```

Comment:
The ```RestrictedFunctionalityException``` is intended for situations where the user needs to buy or upgrade or something like that. In here it is thrown for every exception that may occur. That seems like a wrong mapping of exception types.